test/integration/banListTest.ts

```diff
@@ -408,25 +408,37 @@ describe('Test: should apply bans to the most recently active rooms first', func
         }
         // create some activity in the same order.
         for (const roomId of protectedRooms.slice().reverse()) {
-            await mjolnir.client.sendMessage(roomId, { body: `activity`, msgtype: 'm.text' });
+            await moderator.sendMessage(roomId, { body: `activity`, msgtype: 'm.text' });
             await new Promise(resolve => setTimeout(resolve, 100));
         }
 
         // check the rooms are in the expected order
         for (let i = 0; i < protectedRooms.length; i++) {
-            assert.equal(mjolnir.protectedRoomsByActivity()[i], protectedRooms[i]);
+            assert.equal(mjolnir.protectedRoomsTracker.protectedRoomsByActivity()[i], protectedRooms[i]);
         }
 
-        const badServer = `evil.com`;
         // just ban one server
+        const badServer = `evil.com`;
         const acl = new ServerAcl(serverName).denyIpAddresses().allowServer("*").denyServer(badServer);
-        await createPolicyRule(moderator, banListId, RULE_SERVER, badServer, `Rule ${badServer}`);
+        const aclRooms: any[] = await new Promise(async resolve => {
+            const rooms: any[] = [];
+            this.mjolnir.client.on('room.event', (room: string, event: any) => {
```

Comment:
Shouldn't you unregister this listener? I fear that leaving it registered could somehow interfere with further tests.

Reply:
It shouldn't at all. The same mjolnir / client instance isn't shared between tests.

Reply:
We shouldn't trust tests to do that either